Release/src/http/client/http_client_named_pipe.cpp

```diff
@@ -0,0 +1,466 @@
+/***
+* Copyright (C) Microsoft. All rights reserved.
+* Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+*
+* =+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+*
+* HTTP Library: Client-side APIs.
+*
+* This file contains the implementation for Windows Desktop, based on WinHTTP.
+*
+* For the latest on this and related APIs, please see: https://github.com/Microsoft/cpprestsdk
+*
+* =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+****/
+#include "stdafx.h"
+
+#include "cpprest/http_headers.h"
+#include "http_client_impl.h"
+
+#include "../common/internal_http_helpers.h"
+
+#include <boost/algorithm/string/predicate.hpp>
+
+#ifndef CPPREST_TARGET_XP
+#include <VersionHelpers.h>
+#endif
+
+#undef min
+#undef max
+
+#define CRLF std::string("\r\n")
+
+namespace web
+{
+namespace http
+{
+namespace client
+{
+namespace details
+{
+
+// Helper function to trim leading and trailing null characters from a string.
+static void trim_nulls(utility::string_t &str)
+{
+    size_t index;
+    for (index = 0; index < str.size() && str[index] == 0; ++index);
+    str.erase(0, index);
+    for (index = str.size(); index > 0 && str[index - 1] == 0; --index);
+    str.erase(index);
+}
+
+// Helper function to build error messages.
+static std::string build_error_msg(unsigned long code, const std::string &location)
+{
+    std::string msg(location);
```

Comment:
A lot of code in this file and http_client_winhttp.cpp file is duplicate. Are you planning to move that to a common file. Also i am wondering why are you not getting multiple definition error?

Reply:
These helper classes are currently duplicated (they are private to the cpp file so there are no multiple definition errors). I am planning on doing one cleanup pass and I may do some refactoring, but since those classes are small and simple I 
may just leave them alone.